Add GA4 events to special route format forms

andysellick opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What
Add GA4 tracking to special route formats.

### Special route formats

Example: https://www.gov.uk/find-local-council
Example result: https://www.gov.uk/find-local-council/waverley (county and district)
Example result: https://www.gov.uk/find-local-council/westminster (local authority)

To test locally, modify `controllers/find_local_council_controller.rb`, replacing the `result` method with the below. This will not allow form submissions to work correctly, but you can go directly to the example result pages. Note the comment part way down that allows switching between the two kinds of result page.

```Ruby
def result
  authority_results = {}
  authority_results["local_authorities"] = []
  first_authority = {}
  first_authority["name"] = "Humbridge"
  first_authority["homepage_url"] = "https://www.gov.uk"

  second_authority = {}
  second_authority["name"] = "Windle"
  second_authority["homepage_url"] = "https://www.google.co.uk"

  authority_results["local_authorities"] << first_authority
  authority_results["local_authorities"] << second_authority # uncomment this line to show the alternative result page

  if authority_results["local_authorities"].count == 1
    @authority = authority_results["local_authorities"].first

    render :one_council
  else
    @county = authority_results["local_authorities"].first
    @district = authority_results["local_authorities"].last

    render :district_and_county_council
  end
end
```

## Why
Part of the GA4 migration.

## Visual changes
None.

Trello cards: 
- https://trello.com/c/RF5EB7QT/547-form-events-formsubmit-formerror-formcomplete-and-informationclick-on-specialroute-format

AshGDS commented on app/views/find_local_council/district_and_county_council.html.erb:
```diff
@@ -40,10 +47,19 @@
-              track_label: @county["homepage_url"]
+              track_label: @county["homepage_url"],
+              ga4_link: {
+                event_name: "information_click",
```
The implementation guide for this also has `'action':'information click',`. Should this be added?

andysellick replied:
Added.

AshGDS commented on app/views/find_local_council/one_council.html.erb:
```diff
@@ -18,10 +25,19 @@
-            track_label: @authority["homepage_url"]
+            track_label: @authority["homepage_url"],
+            ga4_link: {
+              event_name: "information_click",
```
The implementation guide also has `'action':'information click',` - should this be added?

andysellick replied:
Added.

AshGDS reviewed (changes requested):
Looks good, I have some small change requests

andysellick commented on app/views/find_local_council/index.html.erb:
```diff
@@ -6,5 +6,6 @@
       publication_format: 'find_local_council',
       postcode: @postcode,
       margin_top: @location_error ? 5 : 0,
+      publication_title: t('formats.local_transaction.find_council'),
```
Thanks, fixed.

andysellick commented on app/views/find_local_council/district_and_county_council.html.erb:
```diff
@@ -75,10 +91,19 @@
+              track_label: @district["homepage_url"],
+              ga4_link: {
+                event_name: "information_click",
+                type: content_item_hash["schema_name"].gsub("_", " "),
```
Thanks, updated. The guide had a comment next to `action` to highlight that the value is the same as `event_name`, I'll check with Harriet.

AshGDS replied:
Cool, no worries

andysellick commented:
@AshGDS think I've addressed your comments if you'd like to check. Also was having some weird merge conflicts trying to rebase the changes into the right commit, so I've merged all the commits into one for simplicity.
